s3po/backends/s3.py

```diff
@@ -107,3 +107,17 @@ def list(self, bucket, prefix=None, delimiter=None, retries=3, headers=None):
         # consume iterator to make a list to keep parity with Swift backend
         return (key.name for key in self._list_retry(retries, bucket,
                                           prefix, delimiter, headers=headers))
+
+    def delete(self, bucket, key, retries, headers=None):
+        '''Delete bucket/key with headers'''
+        # Make our headers object
+        headers = headers or {}
+        bucket = self.get_bucket(bucket)
+        key = bucket.new_key(key)
+
+        @retry(retries)
+        def func():
+            '''The bit that we want to retry'''
+            key.delete(headers=headers)
```

Comment:
I would expect it to, though it would have been nice if the documentation had been clear about it. Tracing through the code confirms that it does, https://github.com/boto/boto/blob/2.36.0/boto/s3/bucket.py#L778, which appears to raise `boto.exceptions.StorageResponseError`. We should catch that and raise `DeleteException` (potentially wrapping the original).